kanidm_unix_int/src/cache.rs

```diff
@@ -264,17 +266,36 @@ impl CacheLayer {
             .map_err(|e| {
                 error!("time conversion error - ex_time less than epoch? {:?}", e);
             })?;
+        // WIP #392: check user/default shell
+        // check if the provided `shell` exists on the system.
+        let mut new_token = token.clone();
+        let user_shell = new_token.shell.unwrap_or_default();
+        let shell_to_use = if Path::new(&user_shell).exists() {
+            // good - the shell exists
+            eprintln!("x392 WIP - OK user_shell <{}> exists.", user_shell);
+            user_shell
+        } else {
+            // bad luck - we have to put our fall back DEFAULT_SHELL
+            eprintln!("x392 WIP - BAD user_shell <{}> missing.", user_shell);
+            // using constants.DEFAULT_SHELL
+            DEFAULT_SHELL.to_string()
+        };
+        eprintln!(
+            "x392 WIP - FIN we will set <{}> as shell_to_use.",
+            shell_to_use
+        );
+        new_token.shell = shell_to_use.into();
 
         let dbtxn = self.db.write().await;
         // We need to add the groups first
-        token
+        new_token
```

Comment:
If you remove the clone, you won't need this change :)

Reply:
I guess I need some help here. I don't see how this can be done without `clone()`. See comment above.